data-raw/healthy-places/travel-time-to-gp.R

```diff
@@ -0,0 +1,233 @@
+library(tidyverse)
+library(geographr)
+library(osmdata)
+library(jsonlite)
+library(httr)
+library(sf)
+library(traveltimeR)
+library(readxl)
+
+Sys.setenv(TRAVELTIME_ID = "<INSERT YOUR ID HERE>")
+Sys.setenv(TRAVELTIME_KEY = "<INSERT YOUR API KEY HERE>")
+
+# ---- Fetch GPs in Northern Ireland ----
+# Bounding box for Northern Ireland
+ni_bb <-
+  getbb("Northern Ireland")
+
+# Search for GPs
+GPs <- opq(ni_bb, timeout = 1000) %>%
+  add_osm_feature("amenity", "doctors") %>%
+  osmdata_sf()
+
+# Get Local Authorities in Great Britain for the next step
+ni_lad <-
+  boundaries_ltla21 |>
+  filter(str_detect(ltla21_code, "^N"))
+
+# Some GPs are not in Northern Ireland - remove them from the dataset
+ni_GPs <-
+  GPs$osm_points[ni_lad, ] |>
+  select(osm_id)
+
+# - Test plot to check GP locations -
+# ni_lad |>
+#   ggplot() +
+#   geom_sf(
+#     fill = NA,
+#     colour = "black"
+#   ) +
+#   geom_sf(
+#     data = ni_GPs,
+#     inherit.aes = FALSE
+#   )
+
+# Calculate neighbours for each Local Authority
+neighbours <- st_touches(ni_lad)
+
+# Look up the Local Authority each GP is in
+ni_GPs_lad <-
+  st_join(ni_GPs, ni_lad) |>
+  filter(!is.na(ltla21_code)) |>
+  mutate(
+    # Round coords to 3 decimal points to save memory
+    lat = st_coordinates(geometry)[,2] |> round(3),
+    lng = st_coordinates(geometry)[,1] |> round(3)
```

Comment:
Are you sure the rounding is saving memory? I ran some quick tests with `lobstr::obj_size()` and whether the rounding is or isn't included, `ni_GPs_lad` comes out to 75.93 kB in any case.

Reply:
That's very helpful to know. We can remove the rounding in that case.